signer/threshold_ed25519_localsoftsign_signer.go

```diff
@@ -0,0 +1,322 @@
+package signer
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	tmJson "github.com/tendermint/tendermint/libs/json"
+	"gitlab.com/polychainlabs/edwards25519"
+	tsed25519 "gitlab.com/polychainlabs/threshold-ed25519/pkg"
+)
+
+// LocalSoftsignThresholdEd25519Signature implements the interface and signs the message for each local signer.
+// LocalSoftSignThresholdEd25519Signature is the implementation of a soft sign signer at the local level.
+type LocalSoftSignThresholdEd25519Signature struct {
+	// UnimplementedThresholdEd25519Signature // embedding unimplemented ThresholdEd25519Signature
+	PubKeyBytes []byte
+	Key         CosignerKey
+	RsaKey      rsa.PrivateKey
+	Total       uint8
+	Threshold   uint8
+
+	// stores the last sign state for a share we have fully signed
+	// incremented whenever we are asked to sign a share
+	//LastSignState *SignState // TODO lift this to the cosigner
+
+	// signing is thread safe
+	//LastSignStateMutex sync.Mutex // TODO lift this to the cosigner
+
+	// Height, Round, Step -> metadata
+	HrsMeta map[HRSTKey]HrsMetadata
+	Peers   map[int]CosignerPeer
```

Comment:
The peers shouldn't be needed inside the implementation, it can move back up to the `LocalCosigner`. The peer publicKey can be passed as a new parameter into `GetEphemeralSecretPart` and `SetEphemeralSecretPart`

Reply:
Should be done now.